libtenzir/src/catalog.cpp

```diff
@@ -542,10 +543,17 @@ auto catalog(catalog_actor::stateful_pointer<catalog_state> self)
       }
       return self->state().merge(std::move(new_synopses));
     },
-    [self](atom::candidates, const tenzir::query_context& query_context)
+    [self](atom::candidates, tenzir::query_context query_context)
       -> caf::result<catalog_lookup_result> {
       TENZIR_TRACE("{} {}", *self, TENZIR_ARG(query_context));
-      return self->state().lookup(query_context.expr);
+      if (not self->state().accept_queries) {
+        return self->state()
+          .delayed_queries
+          .emplace(self->make_response_promise<catalog_lookup_result>(),
+                   std::move(query_context))
+          .first;
+      }
+      return self->state().lookup(std::move(query_context.expr));
     },
```

Comment:
The actor state is also accessed in other handlers, which are not guarded in a similar way. Do we need to do something about that?

Reply:
Afaict they're only ever accessed through the index actor currently, which already guards against that. But that's non-local knowledge and could easily break, so it'd probably be better to guard them as well.

Reply:
This actually fixed a bug accidentally that I did not know we had: the rebuilder now does its first run immediately after starting the node, and not just after 2h.